### DISCUSSION TITLE
Latching button acting like... latching button

### DISCUSSION THREAD

**Discussion by Wapata** [shanteacontrols/OpenDeck#147] [Ideas]
Me again for something that feel natural to me. I think it's not for you because you use the Midi-controllers with softwares that react to a click of a button (aka midi event) and me inline between my master-keyboard and midi-sound-generator.

So.
We can set momentary or latching button. But they do not act as "mono state" or "double state".

One exemple : lets say I use a physical latching button. It have tree pin, it's two button with inverted states.
So I use it this way : State one = MidiCC66=127 and State two = MidiCC66=1 

But in fact I do not need tree wires for that, because if State one is on so state two is off and if state one is _not_ on so state two is on.
So, with one real physical latching interrupter, i should be able to send two event regarding its state.
Right now, OpenDeck Configurator can't do that. Will it be able to do that ?

I use a trick right now, with and interrupter that use two button input and two pages of OpenDeck Configurator. But by doing that in can't use some other cool effects like using a momentary physical button to act as Expression CC11=1/127 and making a morse keyer
like effect

![c3423_large_miniature_spdt_toggle_switch](https://user-images.githubusercontent.com/15179434/153769819-f3d3d68a-fdb3-4445-88c0-3185d9799dd6.jpg)

**Comment by paradajz** [comment]
Would it be possible for you to use pins 1 and 2 as one button, and pins 2 and 3 as second button?

**Comment by Wapata** [comment]
It is totally possible. It just divide by two the functions usable (as one function is two state and so use two button). But it's working and i do it already.
One thing, OpenDeck read the "rising" or "falling" of a button in real time but not at power-on or preset-change. So the physical button do not show the real state of a function until it's actuated one time.

**Reply by paradajz** [reply]
> One thing, OpenDeck read the "rising" or "falling" of a button in real time but not at power-on or preset-change. So the physical button do not show the real state of a function until it's actuated one time.

Once the PC opens USB connection, a second later all the button states are resent (and also for all enabled analog components). Example on Discovery:

![image](https://user-images.githubusercontent.com/2544094/153773126-684e52e8-0e7a-4c35-9661-747338091071.png)

EDIT: The same also happens when changing the preset. So, I wonder how are you testing this?

**Reply by Wapata** [reply]
I was reading it badly.
Because my button are not set to "none" i've read that as a polluting messages and miss the content :-( 
Now i've cleaned the non used input, it's clear that it work as you show.

**Comment by Wapata** [comment]
Wait...
Can you confirm me the behavior of this ?
![opendeck_003](https://user-images.githubusercontent.com/15179434/153773207-8bd4ef3a-59b3-487e-a965-aaaa162eec17.PNG)

I think it's working a i want, CC=value if button is ON  and CC=0 if button is release ?

**Reply by Wapata** [reply]
Reply to myself and futur users ; yes it work !
https://github.com/shanteacontrols/OpenDeck/wiki/Configurable-features#type-section is to be read more than once !

**Comment by Wapata** [comment]
Conclusions of this topic : 
Double state button can work easily with one wire to the ground, one wire to digital input, and OpenDeck Configurator should be set at _Midi message_ : **CC/0 Off**
It will do CCxx Value = **On velocity** / CCxx Value = 0

Edit : sorry paradajz
